jira/resilientsession.py

```diff
@@ -166,25 +196,27 @@ def __verb(self, verb, url, retry_data=None, **kwargs):
         if exception is not None:
             raise exception
         raise_on_error(response, verb=verb, **kwargs)
+        # after raise_on_error, only Response objects are allowed through
+        response = cast(Response, response)  # tell mypy only Response-like are here
         return response
 
-    def get(self, url, **kwargs):
-        return self.__verb("GET", url, **kwargs)
+    def get(self, url: Union[str, bytes], **kwargs) -> Response:
+        return self.__verb("GET", str(url), **kwargs)
 
-    def post(self, url, **kwargs):
-        return self.__verb("POST", url, **kwargs)
+    def post(self, url: Union[str, bytes], data=None, json=None, **kwargs) -> Response:
+        return self.__verb("POST", str(url), data=data, json=json, **kwargs)
 
-    def put(self, url, **kwargs):
-        return self.__verb("PUT", url, **kwargs)
+    def put(self, url: Union[str, bytes], data=None, **kwargs) -> Response:
+        return self.__verb("PUT", str(url), data=data, **kwargs)
 
-    def delete(self, url, **kwargs):
-        return self.__verb("DELETE", url, **kwargs)
+    def delete(self, url: Union[str, bytes], **kwargs) -> Response:
+        return self.__verb("DELETE", str(url), **kwargs)
 
-    def head(self, url, **kwargs):
-        return self.__verb("HEAD", url, **kwargs)
+    def head(self, url: Union[str, bytes], **kwargs) -> Response:
+        return self.__verb("HEAD", str(url), **kwargs)
 
-    def patch(self, url, **kwargs):
-        return self.__verb("PATCH", url, **kwargs)
+    def patch(self, url: Union[str, bytes], data=None, **kwargs) -> Response:
+        return self.__verb("PATCH", str(url), data=data, **kwargs)
 
-    def options(self, url, **kwargs):
-        return self.__verb("OPTIONS", url, **kwargs)
+    def options(self, url: Union[str, bytes], **kwargs) -> Response:
```

Comment:
I think you could use [AnyStr](https://docs.python.org/3/library/typing.html#typing.AnyStr) for `Union[str, bytes]`.

Reply:
Unfortunately we can't use `AnyStr` in this case. This is because this class is a subclass of `Session`, so the typehints need to be consistent with the parent class. Technically we could add a `# type: ignore[override]`, but I felt it unnecessary.


#### Example of mypy failure
```suggestion
    def options(self, url: AnyStr, **kwargs) -> Response:
```

```
jira\resilientsession.py:221: error: Signature of "options" incompatible with supertype "Session"
Found 1 error in 1 file (checked 19 source files)
```